cli/src/pcluster/templates/pcluster_dashboard_builder.py

```diff
@@ -0,0 +1,586 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from aws_cdk import aws_cloudwatch as cloudwatch
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import core
+from aws_cdk.aws_cloudwatch import GraphWidget, LogQueryWidget, TextWidget
+
+from pcluster.models.cluster_config import BaseClusterConfig, SharedStorageType
+
+
+class PclusterDashboardConstruct(core.Construct):
+    """Create the resources required when creating cloudwatch dashboard."""
+
+    def __init__(
+        self,
+        scope: core.Construct,
+        id: str,
+        stack_name: str,
+        cluster_config: BaseClusterConfig,
+        head_node_instance: ec2.CfnInstance,
+        storage_resource_ids,
+        cw_logging_enabled: bool,
+        dcv_enable: bool,
+        **kwargs,
+    ):
+        super().__init__(scope, id)
+        self.stack_scope = scope
+        self.stack_name = stack_name
+        self.config = cluster_config
+        self.head_node_instance = head_node_instance
+        self.dashboard_name = self.stack_name + self._stack_region
+        self._storage_resource_ids = storage_resource_ids
+        self.cw_logging_enabled = cw_logging_enabled
+        self.dcv_enable = dcv_enable
+        self.coord = {"x": 0, "y": 0}
+        self.graph_width = 6
+        self.graph_height = 6
+        self.logs_width = 24
+        self.logs_height = 6
+        self.empty_section = {"bool": True}
```

Comment:
Done. Set `self.empty_section = True`. I was using this because python dict has a method update() that updates the key with the new value, after thinking for a while, I think it make sense to just reassign its value.